src/metadataGeneration/typeResolver.ts

```diff
@@ -309,7 +309,16 @@ export class TypeResolver {
 
   private getTypeName(typeName: string, genericTypes?: ts.NodeArray<ts.TypeNode>): string {
     if (!genericTypes || !genericTypes.length) { return typeName; }
-    return typeName + genericTypes.map((t) => this.getAnyTypeName(t)).join('');
+
+    const resolvedName = genericTypes.reduce((acc, generic) => {
+      if (ts.isTypeReferenceNode(generic) && generic.typeArguments && generic.typeArguments.length > 0) {
+        return [...acc, this.getTypeName(generic.typeName.getText(), generic.typeArguments)];
```

Comment:
@WoH I mentioned this in another recent PR, but there's a performance hit for using the spread operator since you are iterating through the `acc` array on every loop of the  `reduce`. The implementation of the `reduce` could still be a pure function if it mutates the seed since the seed is not considered to be part of the inputs. In fact, it's much more standard in what I've seen to just use `acc.push`. Now I'm aware that `genericTypes` probably has a very small length, so the performance advantages are probably minimal, but I also think that `.push` is more readable. Here's the JSPerf from last time to demonstrate the performance advantages: https://jsperf.com/reduce-with-new-set-vs-seed

Reply:
Rather than have you make this `acc.push` update @WoH, I wanted to be considerate of the number of excellent contributions you've made. So I've made the update as part of https://github.com/lukeautry/tsoa/pull/428 which fixes the other issue that Asterix presented to us. So I'll merge this PR so I can fold it into mine. Sound good? :)